contracts/account/library.cairo

```diff
@@ -0,0 +1,558 @@
+        assert_guardian_set();
+
+        let (current_escape) = _escape.read();
+        let (block_timestamp) = get_block_timestamp();
+        with_attr error_message("argent: not escaping") {
+            assert_not_zero(current_escape.active_at);
+        }
+        with_attr error_message("argent: escape not active") {
+            assert_le(current_escape.active_at, block_timestamp);
+        }
+        with_attr error_message("argent: escape type invalid") {
+            assert current_escape.type = ESCAPE_TYPE_GUARDIAN;
+        }
+
+        // clear escape
+        let new_escape: Escape = Escape(0, 0);
+        _escape.write(new_escape);
+
+        // change guardian
+        assert_not_zero(new_guardian);
+        _guardian.write(new_guardian);
+        guardian_escaped.emit(new_guardian=new_guardian);
+
+        return ();
+    }
+
+    func escape_signer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        new_signer: felt
+    ) {
+        alloc_locals;
+
+        // only called via execute
+        assert_only_self();
+        // no escape when the guardian is not set
+        assert_guardian_set();
+
+        let (current_escape) = _escape.read();
+        let (block_timestamp) = get_block_timestamp();
+        with_attr error_message("argent: not escaping") {
+            assert_not_zero(current_escape.active_at);
+        }
+        with_attr error_message("argent: escape not active") {
+            assert_le(current_escape.active_at, block_timestamp);
+        }
+        with_attr error_message("argent: escape type invalid") {
+            assert current_escape.type = ESCAPE_TYPE_SIGNER;
+        }
+
+        // clear escape
+        let new_escape: Escape = Escape(0, 0);
+        _escape.write(new_escape);
+
+        // change signer
+        assert_not_zero(new_signer);
+        _signer.write(new_signer);
+        signer_escaped.emit(new_signer=new_signer);
+
+        return ();
+    }
+
+    /////////////////////
+    // VIEW FUNCTIONS
+    /////////////////////
+
+    func is_valid_signature{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, ecdsa_ptr: SignatureBuiltin*, range_check_ptr
+    }(hash: felt, sig_len: felt, sig: felt*) -> (is_valid: felt) {
+        alloc_locals;
+
+        let (is_signer_sig_valid) = validate_signer_signature(hash, sig_len, sig);
+        let (is_guardian_sig_valid) = validate_guardian_signature(hash, sig_len - 2, sig + 2);
+
+        // Cairo's way of doing `&&` is by multiplying the two booleans.
+        return (is_valid=is_signer_sig_valid * is_guardian_sig_valid);
+    }
+
+    func supportsInterface{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        interfaceId: felt
+    ) -> (success: felt) {
+        // 165
+        if (interfaceId == 0x01ffc9a7) {
+            return (TRUE,);
+        }
+        // IAccount
+        if (interfaceId == ERC165_ACCOUNT_INTERFACE) {
+            return (TRUE,);
+        }
+        return (FALSE,);
```

Comment:
It would be the case on L1, but on StarkNet `is_valid_signature` is part of the `IAccount` interface so I don't see the need to specify it individually.

Reply:
@milancermak They have been pushing to use camelCase for all external methods (and keep snake case for internal methods). The reason behind is that they want tokens on StarkNet to follow exactly the ERC20/ERC721 standard which uses camelCase. To not end up with external methods being sometimes camelCase when following an Ethereum standard, and snakeCase otherwise they are pushing to have all external methods camelCase.

While I don't like it and would prefer to use snakeCase everywhere in StarkNet, as you mention it's important that 
 accounts use the exact same casing.
 
 I'm planning to address that in a separate PR and apply this convention to the ArgentAccount.